File: ares-core/src/main/java/dev/tigr/ares/core/feature/module/Category.java
package dev.tigr.ares.core.feature.module;

import java.util.ArrayList;

/**
 * @author Tigermouthbear
 */
public enum Category {
    COMBAT,
    EXPLOIT,
    HUD,
    MOVEMENT,
    PLAYER,
    RENDER,
    MISC;

    private final ArrayList<Module> modules = new ArrayList<>();

    public Category add(Module module) {
        modules.add(module);
        return this;
    }

    public ArrayList<Module> getModules() {
        return modules;
    }
}

File: core/src/main/kotlin/org/aresclient/ares/module/Category.kt
package org.aresclient.ares.module

import org.aresclient.ares.renderer.Texture

enum class Category {
    PLAYER,
    COMBAT,
    MOVEMENT,
    RENDER,
    HUD,
    MISC;

    val prettyName = name.lowercase().replaceFirstChar { it.uppercaseChar() }
    val icon by lazy { Texture(Category::class.java.getResourceAsStream("/assets/ares/textures/icons/categories/${name.lowercase()}.png"), false) }
    val modules = arrayListOf<Module>()
}
